src/test/java/uk/gov/di/ipv/cri/common/library/service/AuditEventFactoryTest.java

```diff
@@ -79,8 +84,29 @@ void shouldCreateAuditEventWithContextAndExtensions() {
         when(mockInstant.getEpochSecond()).thenReturn(timestamp);
         when(mockInstant.toEpochMilli()).thenReturn(eventTimestampMs);
         when(mockClock.instant()).thenReturn(mockInstant);
-        PersonIdentityDetailed personIdentity = mock(PersonIdentityDetailed.class);
-        Map<String, String> requestHeaders = Map.of("X-Forwarded-For", clientIpAddress);
+
+        NamePart firstNamePart = new NamePart();
+        firstNamePart.setType("GivenName");
+        firstNamePart.setValue("Jon");
+        NamePart surnamePart = new NamePart();
+        surnamePart.setType("FamilyName");
+        surnamePart.setValue("Smith");
+        Name name = new Name();
+        name.setNameParts(List.of(firstNamePart, surnamePart));
+        List<Name> names = List.of(name);
+        BirthDate birthDate = new BirthDate();
+        birthDate.setValue(LocalDate.of(1984, 6, 27));
+        List<BirthDate> birthDates = List.of(birthDate);
+        @SuppressWarnings("deprecation")
+        PersonIdentityDetailed personIdentity = new PersonIdentityDetailed(names, birthDates);
+        String encodedDeviceInformation = "123456789";
+
+        Map<String, String> requestHeaders =
+                Map.of(
+                        "X-Forwarded-For",
+                        clientIpAddress,
+                        "txma-audit-encoded",
+                        encodedDeviceInformation);
```

Comment:
Instead of editing the test, should have create a new test specifically for the DeviceInfo scenario